tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/ui/pin/PinUnlockActivity.kt

```diff
@@ -70,12 +72,20 @@ class PinUnlockActivity : BasePinActivity() {
     override fun onPinChange(pinLength: Int, intermediatePin: String?) {
         super.onPinChange(pinLength, intermediatePin)
         pinTopText.setTextColor(ContextCompat.getColor(this, R.color.wa_white))
-        pinTopText.text =  getString(
-            when (returnActivity) {
-                ReturnActivity.SETTINGS.getActivityOrder() -> R.string.LockPinSet_Settings_EnterCurrentPin
-                ReturnActivity.CAMOUFLAGE.getActivityOrder() -> R.string.LockPinSet_Settings_EnterCurrentPinToChangeCamouflage
-                else -> R.string.UnlockPin_Message_EnterPin
-            })
+        when (returnActivity) {
+            ReturnActivity.SETTINGS.getActivityOrder() -> {
+                pinTopText.text = getString(R.string.LockPinSet_Settings_EnterCurrentPin)
+                pinEditText.setHint(R.string.LockPinSet_Settings_EnterCurrentPin)
+            }
+            ReturnActivity.CAMOUFLAGE.getActivityOrder() -> {
+                pinTopText.text = getString(R.string.LockPinSet_Settings_EnterCurrentPinToChangeCamouflage)
```

Comment:
1. Since the code for setting the text and hint values for pinTopText and pinEditText is repeated in each branch of the when statement, you can extract it outside the when statement to eliminate duplication
2. Instead of using the getString() method to retrieve the string resource, you can use the resource ID directly when setting the text and hint values.

I know this is an old code, but just some comments to improve it. :)

```
pinTopText.setTextColor(ContextCompat.getColor(this, R.color.wa_white))

val pinTextResId = when (returnActivity) {
    ReturnActivity.SETTINGS.getActivityOrder() -> {
        R.string.LockPinSet_Settings_EnterCurrentPin
    }
    ReturnActivity.CAMOUFLAGE.getActivityOrder() -> {
        R.string.LockPinSet_Settings_EnterCurrentPinToChangeCamouflage
    }
    else -> {
        R.string.UnlockPin_Message_EnterPin
    }
}

pinTopText.setText(pinTextResId)
pinEditText.setHint(pinTextResId)
```